src/runtime/Object.h

```diff
@@ -509,13 +509,17 @@ class ObjectPropertyDescriptor {
 
     bool hasJSGetter() const
     {
-        ASSERT(!isDataProperty());
+        if (!isDataProperty()) {
+            return false;
```

Comment:
It seems that this condition is reversed.
If property is data property type then return false value.
One more thing, would you please add `UNLIKELY` macro to this if statement?

Reply:
IMO you can check this things on EscargotPublic.cpp for public API